python/ray/_private/structured_logging/formatters.py

```diff
@@ -0,0 +1,27 @@
+import logging
+import json
+
+# A dict_keys object containing the standard attributes of a LogRecord. This is used to
+# help us determine which attributes constitute Ray or user-provided context.
+LOGRECORD_STANDARD_ATTRS = logging.makeLogRecord({"message": "test"}).__dict__.keys()
+
+
+class JSONFormatter(logging.Formatter):
+    def format(self, record):
+        record_format = {
+            "ts": self.formatTime(record),
+            "level": record.levelname,
+            "msg": record.getMessage(),
+            "filename": record.filename,
+            "lineno": record.lineno,
+        }
+        if record.exc_info:
+            if not record.exc_text:
+                record.exc_text = self.formatException(record.exc_info)
+            record_format["exc_text"] = record.exc_text
+
+        for key, value in record.__dict__.items():
+            # Both Ray and user-provided context are stored in `record_format`.
+            if key not in LOGRECORD_STANDARD_ATTRS:
```

Comment:
Just to share an idea (doesn't need to be this way), in Serve we have a specific key `SERVE_LOG_EXTRA_FIELDS` that contains all the extra fields a user might want to add. We don't necessarily need a blocklist like this to accept everything outside of `LOGRECORD_STANDARD_ATTRS`. We can ensure the user adds their extra keys into a `RAY_EXTRA` key and just populate json with those keys. 

You can find how that's done in Serve https://github.com/ray-project/ray/blob/e5c80c6f859bb70f367bebed229068721c60ffb7/python/ray/serve/_private/logging_utils.py#L119

Reply:
I think this may increase the context required for users who want to use this feature. Users will need to know about `RAY_EXTRA` and rewrite their logging function calls to use the feature. For example, they need to rewrite their existing function call from `logger.info("Driver process", extra={"username": "johndoe"})` to `logger.info("Driver process", extra={RAY_EXTRA: {"username": "johndoe"}})`. 

However, if Ray Serve has already supported `RAY_EXTRA`, it is better for Ray Core to support the same convention.

Reply:
Update: Sync offline. We decide not to use `RAY_EXTRA` to ensure the Ray logger has the same behavior as the Python's logger.